### DISCUSSION TITLE
How to pop_stream_data_from_stream_buffer from a specific stream puffer ?

### DISCUSSION THREAD

**Discussion by VietnameseRick** [oliver-zehentleitner/unicorn-binance-websocket-api#295] [Q&A] [answered]
How to pop_stream_data_from_stream_buffer from a specific stream puffer ? Also, are there any way to name/label so it will be easier to choose what stream to pop data from ?

**Comment by VietnameseRick** [comment]
@oliver-zehentleitner hi oliver, can you please help me out ...

**Comment by oliver-zehentleitner** [comment, marked as answer]
Its pretty easy ([docs](https://unicorn-binance-websocket-api.docs.lucit.tech/unicorn_binance_websocket_api.html?highlight=stream_buffer_name#unicorn_binance_websocket_api.manager.BinanceWebSocketApiManager.pop_stream_data_from_stream_buffer)):
```
ubwa.create_stream(channel, markets, stream_buffer_name="buffer_1")
oldest_stream_data_from_stream_buffer = ubwa.pop_stream_data_from_stream_buffer("buffer_1")
```

Here is an example file:
https://github.com/LUCIT-Systems-and-Development/unicorn-binance-websocket-api/blob/master/example_stream_buffer_extended.py

**Reply by VietnameseRick** [reply]
Thank you 👍
